include/rovio/ImagePyramid.hpp

```diff
@@ -59,6 +61,45 @@ inline void halfSample(const cv::Mat& imgIn,cv::Mat& imgOut){
   }
 }
 
+template <int n_levels> class ImagePyramidMask {
+public:
+  ImagePyramidMask(){};
+  virtual ~ImagePyramidMask(){};
+  cv::Mat masks_[n_levels]; /**<Array, containing the pyramid masks.*/
+
+  // Computes image masks for the whole image pyramid.
+  // Check fails if mask is an empty cv::Mat
+  void computeFromMask(const cv::Mat &mask, const bool useCv = false) {
+    CHECK(!mask.empty()) << "Image mask is invalid!";
+    mask.copyTo(masks_[0]);
+    for (int i = 1; i < n_levels; ++i) {
+      if (!useCv) {
+        halfSample(masks_[i - 1], masks_[i]);
+      } else {
+        cv::pyrDown(masks_[i - 1], masks_[i],
```

Comment:
This should explicitly use `CV_INTER_LINEAR` for downsampling, otherwise the sub-sampled masks will have soft edges.

Reply:
Thanks, good idea.